WordPress/Classes/Services/ReaderSearchSuggestionService.swift

```diff
@@ -30,71 +40,35 @@ import CocoaLumberjack
+            suggestions = try context.fetch(fetchRequest) as! [ReaderSearchSuggestion]
         } catch let error as NSError {
             DDLogError("Error fetching search suggestion for phrase \(phrase) : \(error.localizedDescription)")
         }
 
         return suggestions.first
     }
 
-
-    /// Finds and returns all ReaderSearchSuggestion starting with the specified search phrase.
-    ///
-    /// - Parameters:
-    ///     - phrase: The search phrase in question.
-    ///
-    /// - Returns: An array of matching `ReaderSearchSuggestion`s.
-    ///
-    @objc func fetchSuggestionsLikePhrase(_ phrase: String) -> [ReaderSearchSuggestion] {
-        let fetchRequest = NSFetchRequest<NSFetchRequestResult>(entityName: "ReaderSearchSuggestion")
-        fetchRequest.predicate = NSPredicate(format: "searchPhrase BEGINSWITH[cd] %@", phrase)
-
-        let sort = NSSortDescriptor(key: "date", ascending: false)
-        fetchRequest.sortDescriptors = [sort]
-
-        var suggestions = [ReaderSearchSuggestion]()
-        do {
-            suggestions = try managedObjectContext.fetch(fetchRequest) as! [ReaderSearchSuggestion]
-        } catch let error as NSError {
-            DDLogError("Error fetching search suggestions for phrase \(phrase) : \(error.localizedDescription)")
-        }
-
-        return suggestions
-    }
-
-
-    /// Deletes the specified search suggestion.
-    ///
-    /// - Parameters:
-    ///     - suggestion: The `ReaderSearchSuggestion` to delete.
-    ///
-    @objc func deleteSuggestion(_ suggestion: ReaderSearchSuggestion) {
-        managedObjectContext.delete(suggestion)
-        ContextManager.sharedInstance().saveContextAndWait(managedObjectContext)
-    }
-
-
     /// Deletes all saved search suggestions.
     ///
     @objc func deleteAllSuggestions() {
-        let fetchRequest = NSFetchRequest<NSFetchRequestResult>(entityName: "ReaderSearchSuggestion")
-        var suggestions = [ReaderSearchSuggestion]()
-        do {
-            suggestions = try managedObjectContext.fetch(fetchRequest) as! [ReaderSearchSuggestion]
-        } catch let error as NSError {
-            DDLogError("Error fetching search suggestion : \(error.localizedDescription)")
-        }
-        for suggestion in suggestions {
-            managedObjectContext.delete(suggestion)
+        self.coreDataStack.performAndSave { context in
+            let fetchRequest = NSFetchRequest<NSFetchRequestResult>(entityName: "ReaderSearchSuggestion")
+            var suggestions = [ReaderSearchSuggestion]()
+            do {
+                suggestions = try context.fetch(fetchRequest) as! [ReaderSearchSuggestion]
+            } catch let error as NSError {
+                DDLogError("Error fetching search suggestion : \(error.localizedDescription)")
+            }
+            for suggestion in suggestions {
+                context.delete(suggestion)
+            }
```

Comment:
Nitpick: Have you considered moving the `for` loop inside the `do catch` to avoid declaring a `var`?

```suggestion
            do {
                try (context.fetch(fetchRequest) as! [ReaderSearchSuggestion]).forEach { context.delete($0) }
            } catch let error as NSError {
                DDLogError("Error fetching search suggestion : \(error.localizedDescription)")
            }
```

Reply:
👍  Addressed in https://github.com/wordpress-mobile/WordPress-iOS/pull/20010/commits/2b5f7d0359141343309f8030de0db284c6e3b6e5